src/llvm-late-gc-lowering.cpp

```diff
@@ -1156,10 +1156,9 @@ bool LateLowerGCFrame::CleanupIR(Function &F) {
                 (gc_use_func != nullptr && callee == gc_use_func)) {
                 /* No replacement */
             } else if (pointer_from_objref_func != nullptr && callee == pointer_from_objref_func) {
-                auto *ASCI = new AddrSpaceCastInst(CI->getOperand(0),
-                    CI->getType(), "", CI);
-                ASCI->takeName(CI);
-                CI->replaceAllUsesWith(ASCI);
+                auto *ptr = new PtrToIntInst(CI->getOperand(0), CI->getType(), "", CI);
```

Comment:
This isn't valid by IR invariants (it'll complain on LLVM 5.0+). Why does this need to be an int?

Reply:
That's what CI->getType() is. We can worry about LLVM 5 when we start supporting it.

Reply:
@yuyichao is running in that configuration AFAIK.

Reply:
I think 4.0 already has non-integral pointer so this will complain there already.

I'm on 4.0 but I do test svn from time to time to make sure it's ready whenever Arch's package is updated.

Should be enough to just keep the old `addrspacecast` and add an `ptrtoint` on top?

Reply:
This comment is not addressed and it breaks LLVM 4.0 .